configure.ac

```diff
@@ -1241,128 +1263,108 @@ else
 fi
 AC_SUBST(PC_NAME_DBUS)
 
-if test "x${enable_systemd}" != xno; then
-    if test $HAVE_dbus = 0; then
-        if test "x${enable_systemd}" = xyes; then
-            AC_MSG_FAILURE([Cannot support systemd resources without DBus])
-        else
-            enable_systemd=no
-        fi
-    fi
-    if test $(echo "$CPPFLAGS" | grep -q PCMK_TIME_EMERGENCY_CGT) \
-            || test "x$ac_cv_have_decl_CLOCK_MONOTONIC" = xno; then
-        if test "x${enable_systemd}" = xyes; then
-            AC_MSG_FAILURE([Cannot support systemd resources without clock_gettime(CLOCK_MONOTONIC, ...)])
-        else
-            enable_systemd=no
-        fi
-    fi
-    if test "x${enable_systemd}" = xtry; then
-        AC_MSG_CHECKING([for systemd version (using dbus-send)])
-        ret=$({ dbus-send --system --print-reply \
-                    --dest=org.freedesktop.systemd1 \
-                    /org/freedesktop/systemd1 \
-                    org.freedesktop.DBus.Properties.Get \
-                    string:org.freedesktop.systemd1.Manager \
-                    string:Version 2>/dev/null \
-                || echo "version unavailable"; } | tail -n1)
-        # sanitize output a bit (interested just in value, not type),
-        # ret is intentionally unenquoted so as to normalize whitespace
-        ret=$(echo ${ret} | cut -d' ' -f2-)
-        AC_MSG_RESULT([${ret}])
-        if test "x${ret}" != xunavailable \
-           || systemctl --version 2>/dev/null | grep -q systemd; then
-            enable_systemd=yes
-        else
-            enable_systemd=no
-        fi
-    fi
-fi
-
+AS_CASE([$enable_systemd],
+        [1], [
```

Comment:
I wonder if, for future readability reasons, it might make sense to define some constants for things like this - 0 = $NO, 1 = $YES.  Maybe something for the "-gt 1" style tests, too.

Reply:
I went with DISABLED/REQUIRED/OPTIONAL, added new commit on top